packages/salesforcedx-vscode-core/src/commands/util/libraryCommandlet.ts

```diff
@@ -5,46 +5,64 @@
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import * as vscode from 'vscode';
+import { channelService } from '../../channels';
+import { nls } from '../../messages';
+import { notificationService } from '../../notifications';
 import {
-  Measurements,
-  Properties,
-  TelemetryData,
+  TelemetryBuilder,
   telemetryService
 } from '../../telemetry';
 import { CommandletExecutor } from './sfdxCommandlet';
 
 export abstract class LibraryCommandletExecutor<T>
   implements CommandletExecutor<T> {
-  protected showChannelOutput = true;
-  protected executionName: string = '';
-  protected startTime: [number, number] | undefined;
-  protected telemetryName: string | undefined;
-
-  public async build(
-    execName: string,
-    telemetryLogName: string
-  ): Promise<void> {}
-
-  public async execute(response: ContinueResponse<T>): Promise<void> {}
-
-  public logMetric(properties?: Properties, measurements?: Measurements) {
-    telemetryService.sendCommandEvent(
-      this.telemetryName,
-      this.startTime,
-      properties,
-      measurements
-    );
-  }
+  protected readonly telemetry = new TelemetryBuilder();
+  protected readonly revealChannelOutput = false;
+  protected abstract readonly logName: string;
+  protected abstract readonly executionName: string;
 
-  public setStartTime() {
-    this.startTime = process.hrtime();
-  }
+  protected abstract run(response: ContinueResponse<T>): Promise<boolean>;
```

Comment:
Non-CLI commands implement the `run` abstract method and return a `boolean` on whether or not the execution was successful.